File: example/src/main/java/ru/virgil/example/system/IdentifiedEntity.java
package ru.virgil.example.system;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;
import java.util.UUID;

@MappedSuperclass
@EqualsAndHashCode(callSuper = true)
@Data
public abstract class IdentifiedEntity extends BaseEntity {

    @GeneratedValue
    @Id
    private UUID uuid;

}

File: example/src/main/java/ru/virgil/example/system/entity/IdentifiedEntity.kt
package ru.virgil.example.system.entity

import java.util.*
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.MappedSuperclass

// TODO: Переместить в базовый модуль
@MappedSuperclass
abstract class IdentifiedEntity(
    @GeneratedValue
    @Id
    var uuid: UUID = UUID.randomUUID(),
) : BaseEntity()
